src/state/sagas/runner/StopRunner/StopRunner.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import { t } from 'i18next';
+import { put, takeEvery, call, select } from 'redux-saga/effects';
+import { Api } from '../../../../services/config/Api';
+import { INGESTION_STATUS } from '../../../../services/config/ApiConstants';
+import { DATASET_ACTIONS_KEY } from '../../../commons/DatasetConstants';
+import { RUNNER_ACTIONS_KEY } from '../../../commons/RunnerConstants';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+
+const getWorkspaceId = (state) => state.workspace.current?.data?.id;
```

Comment:
IIUC, we pass the `workspaceId` via the hook, do we really need to recalculate it in the saga ? :thinking:

Reply:
Right, it is a mess...